File: redsheet-portal/src/app/database-models/v-event-planning-action-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vEventPlanningActionRecord {
    constructor(vEventPlanningActionRecord?: vEventPlanningActionRecord )  {
        let properties = [ "eventPlanningActionBy", "eventPlanningActionDetail", "eventPlanningActionId", "eventPlanningActionOn", "projectPreparationId",];
        for (let property of properties)
            if (vEventPlanningActionRecord && vEventPlanningActionRecord[property])
                this[property] = vEventPlanningActionRecord[property];
    }
 
    @required()
    @maxLength(1000)
	eventPlanningActionBy : string =   undefined;
 
    @required()
    @maxLength(1000)
	eventPlanningActionDetail : string =   undefined;
 
	eventPlanningActionId : number =   0 ;
 
    @required()
	eventPlanningActionOn : Date =   undefined;
 
    @range(1,2147483647)
	projectPreparationId : number =   undefined;


}

File: RedSheet.DbEntities/Models/vEventPlanningActionRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vEventPlanningActionRecords", Schema = "dbo")]
	[Serializable]
	public partial class vEventPlanningActionRecord  
	{
 
	[Required] 
	[MaxLength(1000)]
	public string EventPlanningActionBy { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string EventPlanningActionDetail { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int EventPlanningActionId { get; set; } 
 
	[Required]
	public System.DateTime EventPlanningActionOn { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectPreparationId { get; set; } 

	public vEventPlanningActionRecord()
	{
    }

    }
}

